### PR TITLE
[Query] Add common socket functions

### PR DESCRIPTION
**Pull request by gichan-jang**
This is the first step for the tensor query.
Add common socket functions for tensor-query.
Other common functions are added later.
tensor_query_src, tensor_query_sink and tensor_query_filter will use these functions.

Signed-off-by: Gichan Jang <gichan2.jang@samsung.com>

**Self evaluation:**
1. Build test: [ *]Passed [ ]Failed [ ]Skipped
2. Run test: [ ]Passed [ ]Failed [*]Skipped

### CODE REVIEW

File: gst/nnstreamer/tensor_query/tensor_query_common.c [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+/** @todo change max read size or need payload */
+#define MAX_READ_SIZE 4096
+
+/**
+ * @brief Create requested socket.
+ * @param[in] hostname the hostname.
+ * @param[in] port a port number.
+ * @param[in] cancellable (nullable) a #GCancellable, or %NULL.
+ * @return Newly created socket or NULL on error.
+ * @note Caller is responsible for unreffiring the returned object with g_object_unref().
```
Comment (kimjh12) [inline gst/nnstreamer/tensor_query/tensor_query_common.c:30]:
```suggestion
 * @note Caller is responsible for unreferring the returned object with g_object_unref().
```
looks like a typo

File: gst/nnstreamer/tensor_query/tensor_query_common.c [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+ */
+GSocket *
+gst_tensor_query_socket_new (const gchar * hostname, guint16 port,
+    GCancellable * cancellable)
+{
+  GSocket *socket;
+  GError *err = NULL;
+  GInetAddress *addr;
+  GSocketAddress *saddr;
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_query/tensor_query_common.c:39]: Move ```GResolver *resolver;``` to Line 46. It's not referred outside.

**Review by myungjoo — APPROVED** [review]
LGTM in overall.

File: gst/nnstreamer/tensor_query/tensor_query_common.c (hunk 1 above)
Comment (gichan-jang) [reply gst/nnstreamer/tensor_query/tensor_query_common.c]: Thanks! Updated.

File: gst/nnstreamer/tensor_query/tensor_query_common.c (hunk 2 above)
Comment (gichan-jang) [reply gst/nnstreamer/tensor_query/tensor_query_common.c:39]: Moved. Thanks!

**Review by kimjh12 — APPROVED** [review]
LGTM!

File: gst/nnstreamer/tensor_query/tensor_query_common.c [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        nns_logd ("gst_tensor_query_socket_new: Cancelled name resolval");
+      } else {
+        nns_loge ("Failed to resolve host '%s': %s", hostname, err->message);
+      }
+      g_clear_error (&err);
+      g_object_unref (resolver);
+      return NULL;
+    }
+
+    addr = G_INET_ADDRESS (g_object_ref (results->data));
```
Comment (wooksong) [inline gst/nnstreamer/tensor_query/tensor_query_common.c:62]: ```results``` looks a Glist of GInetAddress. Then, is it guaranteed that there is only one element in this GList?

File: gst/nnstreamer/tensor_query/tensor_query_common.h [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+ * @return Newly created socket or NULL on error.
+ * @note Caller is responsible for unreffiring the returned object with g_object_unref().
+ */
+extern GSocket *
+gst_tensor_query_socket_new (const gchar * hostname, guint16 port,
+    GCancellable * cancellable);
+/**
+ * @brief Receive data from a socket.
+ * @param[in] socket the socket.
+ * @param[in] cancellable (nullable) GCancellable or NULL.
```
Comment (wooksong) [inline gst/nnstreamer/tensor_query/tensor_query_common.h:49]: One of either ```(nullable)``` and ```NULL``` would be enough.

File: gst/nnstreamer/tensor_query/tensor_query_common.h [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+ * @param[in/out] bytes_received Add the number of bytes received.
```
Comment (wooksong) [inline gst/nnstreamer/tensor_query/tensor_query_common.h:50]: Add?

File: gst/nnstreamer/tensor_query/tensor_query_common.c (hunk 3 above)
Comment (gichan-jang) [reply gst/nnstreamer/tensor_query/tensor_query_common.c:61]: Right. results have more than one address.
But I think checking all addresses is over-engineering at this stage, as this route is also a defense code.
I marked it as @todo and make up for it next time.

**Pull request merged**